resources/intro-to-observables.md

```diff
@@ -0,0 +1,123 @@
+
+Ok, so that definition is pretty wordy and technical. 
+
+In simple terms, the pattern involves an object (called an observable) that will share (or emit) data as it arrives over time. Throughout your code, many different objects (called observers) can sign up to be notified when that data arrives (called subscribing). So, whenever our observable recieves its data, it will broadcast that information out and the observers will be listening for it.  
+
+Some key points to note about the pattern:
+
+- You can have one observable and many observers (a one-to-many dependancy). That is, one broadcaster and an unlimited number of recipients.
+- When that observable changes, all observers should be updated/notified immediately and automatically
+- It is the observer's responsibility to register and unregister themselves from the observable (to be notified of changes) and to update their own state when notified.
+
+## So... an Observable?
+
+Think of an observable as a box that you create to hold data from a service or function whose data or items will arrive over time. 
+
+As the data arrives, the box notifies anyone that has requested they be updated and it does so in the way they've asked. (Note: once new data arrives, the previous data no longer exists in the box. It's up to the listener to maintain the previous data if needed.)  
+
+## Basic structure
+
+In its simpliest form, an Observable is a function that accepts an object (an observer) with next, error and complete methods on it. 
+
+When you create an observable, you call its constructor and pass in a function. That function takes in an observer as a parameter and its body contains instructions for the observable to know when to call each of the observer's methods. 
+
+For example ([source](https://medium.com/@benlesh/learning-observable-by-building-observable-d5da57405d87))
+
+```javascript
+const this.data = Observable.create(observer => {
+    const datasource = new DataSource();
+    datasource.ondata = (e) => observer.next(e);
+    datasource.onerror = (err) => observer.error(err);
+    datasource.oncomplete = () => observer.complete();
+    return () => {
+        datasource.destroy();
+    };
+});
+```
+
+The newly created observable doesn't actually do anything until: 
+- a) someone subscribes and passes in an observer and 
+- b) data is emitted.
+
+When you subscribe to an observable, this is the equivilant to calling the observable like a function and passing in an observer. This tells the observerable that you'd like to be notified when data arrives, and tells it which functions to run (via the observer object) to do that.
+
+```javascript
+this.data
+    .subscribe(myObserver);
```

Comment:
this myObserver variable is down below - should I move that up here also, so you get the context?